packages/flutter_tools/lib/src/cache.dart

```diff
@@ -105,21 +107,40 @@ class Cache {
     }
   }
 
-  static String _dartSdkVersion;
+  String _dartSdkVersion;
 
-  static String get dartSdkVersion => _dartSdkVersion ??= platform.version;
+  String get dartSdkVersion => _dartSdkVersion ??= platform.version;
 
-  static String _engineRevision;
+  String _engineRevision;
 
-  static String get engineRevision {
-    if (_engineRevision == null) {
-      final File revisionFile = fs.file(fs.path.join(flutterRoot, 'bin', 'internal', 'engine.version'));
-      if (revisionFile.existsSync())
-        _engineRevision = revisionFile.readAsStringSync().trim();
-    }
+  String get engineRevision {
+    _engineRevision ??= getVersionFor('engine');
     return _engineRevision;
   }
 
+  String _engineDartVersion;
+
+  String get engineDartVersion {
+    if (_engineDartVersion == null) {
+      final Directory engineDirectory = getArtifactDirectory('engine');
+      File dartSdkBin;
+      List<String> dartSdkBinParts;
+      if (platform.isLinux) {
+        dartSdkBinParts = <String>['linux-x64', 'dart-sdk', 'bin', 'dart'];
+      } else if (platform.isMacOS) {
+        dartSdkBinParts = <String>['darwin-x64', 'dart-sdk', 'bin', 'dart'];
+      } else if (platform.isWindows) {
+        dartSdkBinParts = <String>['windows-x64', 'dart-sdk', 'bin', 'dart.exe'];
+      }
+      dartSdkBin = engineDirectory.childFile(fs.path.joinAll(dartSdkBinParts));
+      if (dartSdkBin != null) {
+        final ProcessResult result = processManager.runSync(<String>[dartSdkBin.path, '--version']);
+        _engineDartVersion = result.stderr.trim().replaceAll('Dart VM version: ', '');
```

Comment:
`stderr`?? That's weird...  We should file a bug against the SDK to print that to `stdout`

Reply:
https://github.com/dart-lang/sdk/issues/31481